### ISSUE TITLE
eventual doesn't build when targeting a 32-bit architecture

### ISSUE THREAD

**Issue by jgallagher** [carllerche/eventual#20]
I believe the cause is using `AtomicU64` from syncbox, which [isn't implemented on non-64-bit platforms](http://carllerche.github.io/syncbox/src/syncbox/atomic.rs.html#244-255). Could AtomicUsize be used instead, perhaps?

Specific build errors:

```
eventual/src/core.rs:794:31: 794:45 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `new`
eventual/src/core.rs:794         AtomicState { atomic: AtomicU64::new(initial.as_u64()) }
                                                       ^~~~~~~~~~~~~~
eventual/src/core.rs:799:31: 799:45 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `new`
eventual/src/core.rs:799         AtomicState { atomic: AtomicU64::new(initial.as_u64()) }
                                                       ^~~~~~~~~~~~~~
eventual/src/core.rs:803:31: 803:42 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `load`
eventual/src/core.rs:803         let val = self.atomic.load(order);
                                                       ^~~~~~~~~~~
eventual/src/core.rs:808:31: 808:82 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `compare_and_swap`
eventual/src/core.rs:808         let ret = self.atomic.compare_and_swap(old.as_u64(), new.as_u64(), order);
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
eventual/src/core.rs:813:21: 813:55 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `fetch_add`
eventual/src/core.rs:813         self.atomic.fetch_add(CONSUMING_MASK, Relaxed);
                                             ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
eventual/src/core.rs:817:31: 817:65 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `fetch_sub`
eventual/src/core.rs:817         let val = self.atomic.fetch_sub(CONSUMING_MASK, Relaxed);
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
eventual/src/core.rs:845:31: 845:65 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `fetch_sub`
eventual/src/core.rs:845         let val = self.atomic.fetch_sub(PRODUCING_MASK, Relaxed);
                                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
eventual/src/select.rs:305:31: 305:45 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `new`
eventual/src/select.rs:305         AtomicState { atomic: AtomicU64::new(init.as_u64()) }
                                                         ^~~~~~~~~~~~~~
eventual/src/select.rs:309:33: 309:44 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `load`
eventual/src/select.rs:309         State::load(self.atomic.load(order))
                                                           ^~~~~~~~~~~
eventual/src/select.rs:313:34: 313:85 error: type `syncbox::atomic::types::AtomicU64` does not implement any method in scope named `compare_and_swap`
eventual/src/select.rs:313         let actual = self.atomic.compare_and_swap(old.as_u64(), new.as_u64(), order);
                                                            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by carllerche** [comment]
Odd. I thought I did it... I will look at this soon.

**Comment by carllerche** [comment]
Sorry about the delay. Fixed here: 0bf38f37dbd0ede059f6f9141a23954d348489ee
